### PR TITLE
website: add name attribute to form component code examples

### PR DESCRIPTION
**Pull request by shleewhite**
### :pushpin: Summary

If merged, the code sections for Form components will no longer have warnings that the code samples are incomplete and instead all of the code examples will have the name attribute.

### :hammer_and_wrench: Detailed description

* removed warnings from: Checkbox, Masked Input, Radio, Select, TextInput, Textarea, Toggle pages
* added the name attribute to all code examples on Checkbox, FileInput, Masked Input, Radio, Select, TextInput, Textarea, Toggle pages

### :camera_flash: Screenshots

**Example of the banners that were removed from form docs pages:**
Screenshot 2024-07-24 at 4 13 43 PM

### :link: External links

Jira ticket: [HDS-2283](https://hashicorp.atlassian.net/browse/HDS-2283)

***

### 👀 Component checklist

- [ ] Percy was checked for any visual regression
- [ ] A changelog entry was added via [Changesets](https://github.com/changesets/changesets) if needed (see [templates here](https://github.com/hashicorp/design-system/blob/main/wiki/Website-Changelog.md#templates-for-npm-packages))

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

[HDS-2283]: https://hashicorp.atlassian.net/browse/HDS-2283?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: website/docs/components/form/checkbox/partials/code/how-to-use.md [hunk 1]
```diff
@@ -50,7 +43,7 @@ The `@name` argument offers an easy way to provide the same name for all the Che
 To better fit your spacing requirements, choose between two different layout orientations: `vertical` or `horizontal`.
 
 ```handlebars
-<Hds::Form::Checkbox::Group as |G|>
+<Hds::Form::Checkbox::Group @name="datacenter" as |G|>
```
Comment (didoo) [inline website/docs/components/form/checkbox/partials/code/how-to-use.md:46]: Suggestion: consider prefixing all the `@name`s with `demo-` so it's more obvious to our consumers that they have to update that value according to their context.

```suggestion
<Hds::Form::Checkbox::Group @name="demo-datacenter" as |G|>
```

**Review by didoo — COMMENTED** [review]
From a code perspective, everything is fine.

I have a more high-level question though? When our consumers will copy&paste one of these code snippets, they will bring with them, in their codebases I mean, also the `name` attribute. 

Is this name required in their codebases to make their form accessible? 

If yes, why (I'm asking @alex-ju @MelSumner) we didn't make it required using a custom `@name` argument? Also, how come all their forms passed (more or less) the accessibility tests?

If no, why do we want them to have an extra `name` in their form, if it's not needed (and potentially may interfere with their data submission?)

**Comment by shleewhite** [comment]
> From a code perspective, everything is fine.
> 
> I have a more high-level question though? When our consumers will copy&paste one of these code snippets, they will bring with them, in their codebases I mean, also the `name` attribute.
> 
> Is this name required in their codebases to make their form accessible?
> 
> If yes, why (I'm asking @alex-ju @MelSumner) we didn't make it required using a custom `@name` argument? Also, how come all their forms passed (more or less) the accessibility tests?
> 
> If no, why do we want them to have an extra `name` in their form, if it's not needed (and potentially may interfere with their data submission?)

It is definitely required for radio buttons to make the keyboard interaction work.

For the other components, I don't see how it could interfere with data submission...

**Review by alex-ju — APPROVED** [review]
Looking good! 🙌 

One thing that I would like us to change before getting this in is the use of `@name` argument in `::Field` variants. As per our API guidance, I would encourage consumers to use the `name` attribute instead of using the argument.

As a background, we added the `@name` argument to allow consumers to set it at the group level, and potentially override it at field or base level if needed.

**Comment by alex-ju** [comment]
> Is this name required in their codebases to make their form accessible?

there are many nuances here and it depends on the implementation of the form; it's best practice to have a `name` attribute on each field (even when they're not part of a group) so we'd like to encourage consumers to use the attribute, but making it a requirement feels a bit strong. It would also mean a breaking change at this point.

**Comment by shleewhite** [comment]
> One thing that I would like us to change before getting this in is the use of `@name` argument in `::Field` variants. As per our API guidance, I would encourage consumers to use the `name` attribute instead of using the argument.
> 
> As a background, we added the `@name` argument to allow consumers to set it at the group level, and potentially override it at field or base level if needed.

@alex-ju do all the field components have `@name`? I don't see it in the types or in the documentation for the components? I only see it for the ones with groups: checkbox, radio

File: website/docs/components/form/checkbox/partials/code/how-to-use.md [hunk 2]
```diff
@@ -211,7 +204,7 @@ The basic invocation requires a `Label`. This creates:
 - an `<input type="checkbox">` control with an automatically generated `ID` attribute.
 
 ```handlebars
-<Hds::Form::Checkbox::Field as |F|>
+<Hds::Form::Checkbox::Field @name="demo-cost-estimation" as |F|>
```
Comment (alex-ju) [inline website/docs/components/form/checkbox/partials/code/how-to-use.md:207]:
```suggestion
<Hds::Form::Checkbox::Field name="demo-cost-estimation" as |F|>
```

File: website/docs/components/form/checkbox/partials/code/how-to-use.md [hunk 3]
```diff
@@ -221,7 +214,7 @@ The basic invocation requires a `Label`. This creates:
 Pass a `@value` argument to the checkbox input.
 
 ```handlebars
-<Hds::Form::Checkbox::Field @value="enable" as |F|>
+<Hds::Form::Checkbox::Field @value="enable" @name="demo-cost-estimation" as |F|>
```
Comment (alex-ju) [inline website/docs/components/form/checkbox/partials/code/how-to-use.md:217]:
```suggestion
<Hds::Form::Checkbox::Field @value="enable" name="demo-cost-estimation" as |F|>
```

File: website/docs/components/form/checkbox/partials/code/how-to-use.md [hunk 4]
```diff
@@ -231,7 +224,7 @@ Pass a `@value` argument to the checkbox input.
 Use the standard HTML `checked` attribute to mark the input as checked.
 
 ```handlebars
-<Hds::Form::Checkbox::Field checked as |F|>
+<Hds::Form::Checkbox::Field checked @name="demo-cost-estimation" as |F|>
```
Comment (alex-ju) [inline website/docs/components/form/checkbox/partials/code/how-to-use.md:227]:
```suggestion
<Hds::Form::Checkbox::Field checked name="demo-cost-estimation" as |F|>
```

File: website/docs/components/form/checkbox/partials/code/how-to-use.md [hunk 5]
```diff
@@ -241,7 +234,7 @@ Use the standard HTML `checked` attribute to mark the input as checked.
 In addition to the checked and unchecked states, a checkbox can be in an indeterminate state, also referred to as partially checked state. A common use case for this state is when a parent checkbox allows the user to select multiple child checkboxes at once. The visual appearance of the checkbox is determined based on the [`indeterminate` attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Input/checkbox#indeterminate_state_checkboxes).
 
 ```handlebars
-<Hds::Form::Checkbox::Field indeterminate={{true}} as |F|>
+<Hds::Form::Checkbox::Field indeterminate={{true}} @name="demo-all-datacenters" as |F|>
```
Comment (alex-ju) [inline website/docs/components/form/checkbox/partials/code/how-to-use.md:237]:
```suggestion
<Hds::Form::Checkbox::Field indeterminate={{true}} name="demo-all-datacenters" as |F|>
```

File: website/docs/components/form/checkbox/partials/code/how-to-use.md [hunk 6]
```diff
@@ -269,7 +262,7 @@ When helper text is added, the component automatically adds an `aria-describedby
 To indicate a field is invalid, provide an error message using the `Error` contextual component.
 
 ```handlebars
-<Hds::Form::Checkbox::Field as |F|>
+<Hds::Form::Checkbox::Field @name="demo-approve-change" as |F|>
```
Comment (alex-ju) [inline website/docs/components/form/checkbox/partials/code/how-to-use.md:265]:
```suggestion
<Hds::Form::Checkbox::Field name="demo-approve-change" as |F|>
```

File: website/docs/components/form/checkbox/partials/code/how-to-use.md [hunk 7]
```diff
@@ -285,7 +278,7 @@ In this case all the internal references (`id/for/aria-describedby`) between the
 !!!
 
 ```handlebars
-<Hds::Form::Checkbox::Field @id="my-control" as |F|>
+<Hds::Form::Checkbox::Field @id="my-control" @name="demo-cost-estimation" as |F|>
```
Comment (alex-ju) [inline website/docs/components/form/checkbox/partials/code/how-to-use.md:281]:
```suggestion
<Hds::Form::Checkbox::Field @id="my-control" name="demo-cost-estimation" as |F|>
```

File: website/docs/components/form/checkbox/partials/code/how-to-use.md [hunk 8]
```diff
@@ -296,7 +289,7 @@ In this case all the internal references (`id/for/aria-describedby`) between the
 Pass an `@extraAriaDescribedBy` argument to the field to connect one or more extra elements describing the field to the control. This provides extra ID values to the `aria-describedby` attribute of the control, in addition to those automatically generated by the component.
 
 ```handlebars
-<Hds::Form::Checkbox::Field @extraAriaDescribedBy="my-extra-element-ID" as |F|>
+<Hds::Form::Checkbox::Field @extraAriaDescribedBy="my-extra-element-ID" @name="demo-cost-estimation" as |F|>
```
Comment (alex-ju) [inline website/docs/components/form/checkbox/partials/code/how-to-use.md:292]:
```suggestion
<Hds::Form::Checkbox::Field @extraAriaDescribedBy="my-extra-element-ID" name="demo-cost-estimation" as |F|>
```

File: website/docs/components/form/checkbox/partials/code/how-to-use.md [hunk 9]
```diff
@@ -317,7 +310,7 @@ This component supports use of `...attributes`. This means you can use all the s
 Because this component supports use of `...attributes`, you can use all the usual Ember techniques for event handling (e.g., `input`, `change`), validation, etc. 
 
 ```handlebars
-<Hds::Form::Checkbox::Field {{on "change" this.yourOnChangeFunction}} as |F|>
+<Hds::Form::Checkbox::Field @name="demo-cost-estimation" {{on "change" this.yourOnChangeFunction}} as |F|>
```
Comment (alex-ju) [inline website/docs/components/form/checkbox/partials/code/how-to-use.md:313]:
```suggestion
<Hds::Form::Checkbox::Field name="demo-cost-estimation" {{on "change" this.yourOnChangeFunction}} as |F|>
```

**Review by alex-ju — COMMENTED** [review]
> @alex-ju do all the field components have @name? I don't see it in the types or in the documentation for the components? I only see it for the ones with groups: checkbox, radio

yes, that's what I meant, we did not document `@name` on `::Field` variants because we want to encourage the use of `name`. I left more concrete comments on what I would change in this PR to align the examples with our documentation

**Pull request merged**